File: fundamentos/java/ConversorTemperatura.java
import java.util.Locale;
import java.util.Scanner;

/**
 * Conversor simples de temperaturas.
 * 
 * Demonstra leitura de dados, condicionais e formatação de saída.
 */
public class ConversorTemperatura {

    private static double celsiusParaFahrenheit(double celsius) {
        return (celsius * 9 / 5) + 32;
    }

    private static double fahrenheitParaCelsius(double fahrenheit) {
        return (fahrenheit - 32) * 5 / 9;
    }

    public static void main(String[] args) {
        Locale.setDefault(Locale.US);
        Scanner scanner = new Scanner(System.in);

        System.out.println("=== Conversor de Temperaturas ===");
        System.out.print("Converter para (C)elsius ou (F)ahrenheit? ");
        String escala = scanner.next().trim().toLowerCase();

        System.out.print("Digite o valor a ser convertido: ");
        double valor = scanner.nextDouble();

        if (escala.equals("c")) {
            double resultado = fahrenheitParaCelsius(valor);
            System.out.printf("%.1f°F equivalem a %.1f°C%n", valor, resultado);
        } else if (escala.equals("f")) {
            double resultado = celsiusParaFahrenheit(valor);
            System.out.printf("%.1f°C equivalem a %.1f°F%n", valor, resultado);
        } else {
            System.out.println("Opção inválida. Utilize 'C' ou 'F'.");
        }

        scanner.close();
    }
}

File: fundamentos/cpp/conversor_temperatura.cpp
#include <iomanip>
#include <iostream>
#include <string>

// Conversor simples de temperaturas.
// Demonstra leitura de dados, condicionais e formatação de saída.

namespace {

double CelsiusParaFahrenheit(double celsius) {
    return (celsius * 9.0 / 5.0) + 32.0;
}

double FahrenheitParaCelsius(double fahrenheit) {
    return (fahrenheit - 32.0) * 5.0 / 9.0;
}

}  // namespace

int main() {
    std::cout << "=== Conversor de Temperaturas ===\n";
    std::cout << "Converter para (C)elsius ou (F)ahrenheit? ";
    std::string escala;
    std::getline(std::cin, escala);

    std::cout << "Digite o valor a ser convertido: ";
    double valor = 0.0;
    std::cin >> valor;

    std::cout << std::fixed << std::setprecision(1);

    if (escala == "c" || escala == "C") {
        double resultado = FahrenheitParaCelsius(valor);
        std::cout << valor << "°F equivalem a " << resultado << "°C\n";
    } else if (escala == "f" || escala == "F") {
        double resultado = CelsiusParaFahrenheit(valor);
        std::cout << valor << "°C equivalem a " << resultado << "°F\n";
    } else {
        std::cout << "Opção inválida. Utilize 'C' ou 'F'.\n";
    }

    return 0;
}
